ts/components/TextStoryCreator.tsx

```diff
@@ -301,6 +301,29 @@ export function TextStoryCreator({
     );
   }, [isColorPickerShowing, colorPickerPopperRef, colorPickerPopperButtonRef]);
 
+  useEffect(() => {
+    if (!isLinkPreviewInputShowing) {
+      return noop;
+    }
+    return handleOutsideClick(
```

Comment:
Can this cleanup callback be a function outside of useEffect with useCallback ? 
I think that will make this code more readable and cleaner.